doc/user-manual/language/reflection.lagda.rst

```diff
@@ -711,3 +711,34 @@ Example usage:
       defId id-name
 
     unquoteDecl id′ = mkId id′
+
+System Calls
+~~~~~~~~~~~~
+
+It is possible to run system calls as part of a metaprogram, using the ``execTC`` builtin. You can use this feature to implement type providers, or to call external solvers. For instance, the following example calls ``/bin/echo`` from Agda:
+
+.. code-block:: agda
+
+  postulate
+    execTC : (exe : String) (args : List String) (stdIn : String)
+           → TC (Σ Nat (λ _ → Σ String (λ _ → String)))
+
+  {-# BUILTIN AGDATCMEXEC execTC #-}
+
+  macro
+    echo : List String → Term → TC ⊤
+    echo args hole = do
+      (exitCode , (stdOut , stdErr)) ← execTC "echo" args ""
+      unify hole (lit (string stdOut))
+
+  _ : echo ("hello" ∷ "world" ∷ []) ≡ "hello world\n"
+  _ = refl
+
+The ``execTC`` builtin takes three arguments: the basename of the executable (e.g., ``"echo"``), a list of arguments, and the contents of the standard input. It returns a triple, consisting of the exit code (as a natural number), the contents of the standard output, and the contents of the standard error.
+
+It would be ill-advised to allow Agda to make arbitrary system calls. Hence, the feature must be activated by passing the ``--allow-exec`` option, either on the command-line or using a pragma. (Note that ``--allow-exec`` is incompatible with ``--safe``.) Furthermore, Agda can only call executables which are listed in the list of trusted executables, ``~/.agda/executables``. For instance, to run the example above, you must add ``/bin/echo`` to this file::
+
+  # contents of ~/.agda/executables
```

Comment:
This is interpreted as Agda code.
Maybe run `make user-manual-{test|...}` locally?

Reply:
The `make user-manual-test` task fails locally, with:
```
$ make user-manual-test
======================================================================
========================= User manual (test) =========================
======================================================================
bash: /path/to/agda/dist-2.6.2/build/agda-tests/agda-tests: No such file or directory
make: *** [user-manual-test] Error 127
```
I'm not sure how to address this. To test the doc build, I've been running `make html` from `/path/to/agda/doc/user-manual` locally.

How do I get `make user-manual-test` to work?

Reply:
As an aside, I don't usually use the top-level Makefile. I believe it used to depend on a global GHC installation. Currently, I have to pass `--extra-include/lib-dirs` to Stack, in order for it to find libicu, and the Makefile makes this quite difficult.

Reply:
Thanks, @gallais !

Reply:
I have pushed a fix given that it's easy to run that test on my machine.
I think the difference with `make html` is that the test actually tries
to typecheck the Agda code in code blocks (unless they are marked as text
or something else).

Reply:
We're now hitting a bug in `master`? :cry:

Reply:
Wait, what bug?

Reply:
I don't know, the build failed somewhere else. I have pushed another fix.